src/organisms/Trips.js

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect } from 'react';
+import Grid from '@material-ui/core/Grid';
+import BusCard from '../Molecules/BusCard.js';
+import InputFeild from '../Atoms/InputFeild.js';
+import RefreshIcon from '@material-ui/icons/Refresh';
+import Custombutton from '../Atoms/Custombutton.js';
+
+const Trips = (props) => {
+    const trips = props.tripdetails
+    const tempTrips = [...trips]
+    const [tripData, setTripData] = React.useState([]);
+    const [flag, setFlag] = React.useState('')
+    const [company, setCompany] = React.useState('')
+    const [temp] = React.useState(tempTrips);
+
+    useEffect(() => {
+        if (company) {
+            let searchdata = trips.filter((ele) => {
+                if (ele.busdetails.buscompany.companyName === company) {
```

Comment:
have null checks everywhere before you read the object data and you could just return ```ele.busdetails.buscompany.companyName === company```  instead of returning true/false in if conditions